lib/v2/screens/dashboard/wallet_screen.dart

```diff
@@ -24,25 +22,30 @@ class WalletScreen extends StatelessWidget {
   Widget build(BuildContext context) {
     return BlocProvider(
       create: (_) => WalletBloc()..add(const RefreshDataEvent()),
-      child: RefreshIndicator(
-          child: Scaffold(
-            appBar: buildAppBar(context) as PreferredSizeWidget?,
-            body: ListView(
-              // TODO(n13): Use exact measurements from figma
-              children: <Widget>[
-                const SizedBox(height: 15),
-                const TokenCardsWidget(),
-                const SizedBox(height: 20),
-                buildSendReceiveButton(context),
-                const SizedBox(height: 20),
-                walletBottom(context),
-              ],
+      child: BlocBuilder<WalletBloc, WalletState>(
+        builder: (context, state) {
+          return RefreshIndicator(
+            onRefresh: () async {
+              BlocProvider.of<RatesBloc>(context).add(const FetchRates());
+              BlocProvider.of<WalletBloc>(context).add(const RefreshDataEvent());
```

Comment:
now `BlocProvider.of<WalletBloc>(context).add(const RefreshDataEvent());` has a context with a WalletBloc.